internal/controllers/device_data_controller.go

```diff
@@ -30,17 +31,16 @@ type QueryDeviceErrorCodesReq struct {
 }
 
 type QueryDeviceErrorCodesResponse struct {
-	Message string `json:"message"`
+	ErrorCodes []services.ErrorCodesResponse `json:"errorCodes"`
 }
 
 type GetUserDeviceErrorCodeQueriesResponse struct {
 	Queries []GetUserDeviceErrorCodeQueriesResponseItem `json:"queries"`
 }
 
 type GetUserDeviceErrorCodeQueriesResponseItem struct {
-	Codes       []string  `json:"errorCodes"`
-	Description string    `json:"description"`
-	RequestedAt time.Time `json:"requestedAt"`
+	ErrorCodesDescription []services.ErrorCodesResponse `json:"errorCodesDescription"`
```

Comment:
I would do "errorCodeDescriptions` or just `errorCodes`. It feels odd to not have the whole thing be plural.

Reply:
How about `errorCodes` just to match the listing response? I don't see a reason for them to differ.